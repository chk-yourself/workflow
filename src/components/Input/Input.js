import React from 'react';
import './Input.scss';

const Input = ({
  className = '',
  name,
  type,
  value,
  onFocus,
  onChange,
  onInput,
  placeholder,
  isRequired,
  onBlur,
  autoComplete,
  inputRef,
  hideLabel,
  label,
  labelClass,
  onKeyDown
}) => {
  return (
    <>
      {!hideLabel && (
        <label htmlFor={name} className={`input__label ${labelClass}`}>
          {label}
        </label>
      )}
      <input
        className={className}
        id={name}
        name={name}
        type={type}
        value={value}
        onFocus={onFocus}
        onChange={onChange}
        placeholder={placeholder}
        required={isRequired}
        onBlur={onBlur}
        autoComplete={autoComplete}
        ref={inputRef}
        onInput={onInput}
        onKeyDown={onKeyDown}
      />
    </>
  );
};

export default Input;
