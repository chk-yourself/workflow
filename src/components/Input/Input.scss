@import '../../styles/variables.scss';

.input {
  box-sizing: border-box;
  border: 2px solid $grey-300;
  padding: 16px 8px;
  font-size: 16px;
  border-radius: 4px;
  width: 100%;
  max-width: 100%;
  color: $black;
  outline: none;
  height: 48px;
  -webkit-appearance: none;
  &::placeholder {
    color: $grey-600;
  }
  &:active,
  &:focus {
    border-color: $primary;
  }
  &:read-only {
    border-color: transparent;
  }
}

.input__label {
  margin-bottom: 6px;
  display: block;
  font-size: 14px;
  color: $grey-800;
  font-weight: 500;
}

.input__helper {
  margin: 8px 0 0 0;
  font-size: 11px;
  color: $grey-600;
  line-height: 1.4;
}
