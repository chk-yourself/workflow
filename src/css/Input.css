@import url("https://rsms.me/inter/inter.css");
.input {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  border: 2px solid #dee2e6;
  padding: 16px 8px;
  font-size: 16px;
  border-radius: 4px;
  width: 100%;
  max-width: 100%;
  color: #000;
  outline: none;
  height: 48px;
  -webkit-appearance: none;
}

.input::-webkit-input-placeholder {
  color: #868e96;
}

.input:-ms-input-placeholder {
  color: #868e96;
}

.input::-ms-input-placeholder {
  color: #868e96;
}

.input::placeholder {
  color: #868e96;
}

.input:active, .input:focus {
  border-color: #3e7aff;
}

.input:-moz-read-only {
  border-color: transparent;
}

.input:read-only {
  border-color: transparent;
}

.input__label {
  margin-bottom: 6px;
  display: block;
  font-size: 14px;
  color: #343a40;
  font-weight: 500;
}

.input__hint {
  margin: 8px 0 0 0;
  font-size: 11px;
  color: #868e96;
  line-height: 1.4;
}
/*# sourceMappingURL=Input.css.map */