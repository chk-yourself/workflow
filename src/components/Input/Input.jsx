import React, { Component } from 'react';
import { ErrorMessage } from '../Error';
import './Input.scss';

let idCounter = 0;

class Input extends Component {
  static defaultProps = {
    className: '',
    innerRef: null,
    labelClass: '',
    type: 'text',
    isRequired: false,
    isReadOnly: false,
    label: '',
    id: null,
    helper: null,
    helperClass: '',
    isInvalid: false
  };

  state = {
    isFocused: false,
    id: (this.props.id || ++idCounter).toString()
  };

  onFocus = e => {
    const { onFocus } = this.props;
    this.setState(prevState => ({
      isFocused: !prevState.isFocused
    }));
    if (onFocus) {
      onFocus(e);
    }
  };

  onBlur = e => {
    const { onBlur } = this.props;
    this.setState({
      isFocused: false
    });
    if (onBlur) {
      onBlur(e);
    }
  };

  render() {
    const {
      className,
      name,
      type,
      value,
      onChange,
      onFocus,
      onBlur,
      placeholder,
      isRequired,
      autoComplete,
      innerRef,
      label,
      labelClass,
      hint,
      hintClass,
      onKeyDown,
      maxLength,
      minLength,
      isReadOnly,
      id: unusedId,
      validationMessage,
      isInvalid,
      ...rest
    } = this.props;

    const { isFocused, id } = this.state;
    const hintId = hint ? `${id}__hint` : '';
    const validationMessageId = validationMessage ? `${id}__validation` : '';
    return (
      <>
        {label && (
          <label
            htmlFor={id}
            className={`input__label ${labelClass} ${isFocused ? 'is-focused' : ''}`}
          >
            {label}
          </label>
        )}
        <input
          id={id}
          className={`input ${className} ${isInvalid ? 'is-invalid' : ''}`}
          name={name}
          type={type}
          value={value}
          onFocus={this.onFocus}
          onChange={onChange}
          placeholder={placeholder}
          required={isRequired}
          onBlur={this.onBlur}
          autoComplete={autoComplete}
          ref={innerRef}
          onKeyDown={onKeyDown}
          maxLength={maxLength}
          minLength={minLength}
          readOnly={isReadOnly}
          tabIndex={isReadOnly ? -1 : 0}
          aria-invalid={isInvalid}
          aria-describedby={`${hintId} ${validationMessageId}`.trim() || null}
          {...rest}
        />
        {validationMessage && <ErrorMessage id={validationMessageId} text={validationMessage} />}
        {hint && (
          <p id={hintId} className={`input__helper ${hintClass} ${isFocused ? 'is-focused' : ''}`}>
            {hint}
          </p>
        )}
      </>
    );
  }
}

export default Input;
