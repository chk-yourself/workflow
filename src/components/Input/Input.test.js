import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('Input Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Input />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
