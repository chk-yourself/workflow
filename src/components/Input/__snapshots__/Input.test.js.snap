// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
Array [
  "",
  <input
    className="input "
    id={null}
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={false}
    required={false}
    tabIndex={0}
    type="text"
  />,
]
`;
